assembly/src/parsers/io_ops.rs

```diff
@@ -262,63 +284,63 @@ fn parse_constants(op: &Token) -> Result<Vec<Felt>, ParsingError> {
     let param_count = op.num_parts() - param_idx;
 
     // for multiple input parameters, parse & push each one onto the stack in order, then return
-    if param_count > 1 {
-        for param_idx in param_idx..=param_count {
-            let value = parse_element_param(op, param_idx)?;
-            constants.push(value);
-        }
-        return Ok(constants);
-    }
-
-    // for a single input, there could be one value or there could be a series of many hexadecimal
-    // values without separators
-    let param_str = op.parts()[param_idx];
-    if let Some(param_str) = param_str.strip_prefix("0x") {
-        // parse 1 or more hexadecimal values
-        let values = parse_hex_params(op, param_idx, param_str)?;
-        // push each value onto the stack in order
-        for &value in values.iter() {
-            constants.push(value);
-        }
-    } else {
-        // parse 1 decimal value and push it onto the stack
-        let value = parse_decimal_param(op, param_idx, param_str)?;
+    for param_idx in param_idx..=param_count {
+        let value = parse_element_param(op, param_idx)?;
         constants.push(value);
     }
-
     Ok(constants)
 }
 
-fn parse_hex_params(
-    op: &Token,
-    param_idx: usize,
-    param_str: &str,
-) -> Result<Vec<Felt>, ParsingError> {
+fn parse_hex_params(op: &Token, param_str: &str) -> Result<Vec<Felt>, ParsingError> {
+    let param_idx = 1;
     // handle error cases where the hex string is poorly formed
-    let is_single_element = if param_str.len() <= HEX_CHUNK_SIZE {
-        if param_str.len() % 2 != 0 {
-            // parameter string is not a valid hex representation
-            return Err(ParsingError::invalid_param(op, param_idx));
-        }
-        true
-    } else {
-        if param_str.len() % HEX_CHUNK_SIZE != 0 {
-            // hex string doesn't contain a valid number of bytes
-            return Err(ParsingError::invalid_param(op, param_idx));
-        }
-        false
-    };
+    if param_str.len() % HEX_CHUNK_SIZE != 0 {
+        // hex string doesn't contain a valid number of bytes
+        return Err(ParsingError::invalid_param(op, param_idx));
+    }
 
-    // parse the hex string into one or more valid field elements
-    if is_single_element {
-        // parse a single element in hex representation
-        let parsed_param = parse_hex_param(op, param_idx, param_str)?;
-        Ok(vec![parsed_param])
+    // iterate over the multi-value hex string and parse each 8-byte chunk into a valid element
+    (0..param_str.len())
+        .step_by(HEX_CHUNK_SIZE)
+        .map(|i| parse_hex_param(op, param_idx, &param_str[i..i + HEX_CHUNK_SIZE]))
+        .collect()
+}
+
+fn evaluate_type_one(value: Felt) -> Result<Node, ParsingError> {
+    let value_u64 = value.as_int();
```

Comment:
Parsing `Felt`'s is more expensive than parsing `u64` values and we should should avoid converting back and forth via `as_int()` as it is also non-negligible. So, I would change the input parameter to `u64` here.